packages/lib/segment/service.ts

```diff
@@ -243,18 +243,51 @@ export const deleteSegment = async (segmentId: string): Promise<TSegment> => {
       select: selectSegment,
     });
 
-    // pause all the running surveys that are using this segment
     const surveyIds = segment.surveys.map((survey) => survey.id);
     if (!!surveyIds?.length) {
-      await prisma.survey.updateMany({
-        where: {
-          id: { in: surveyIds },
-          status: "inProgress",
-        },
-        data: {
-          status: "paused",
-        },
-      });
+      for (const surveyId of surveyIds) {
```

Comment:
It seems like we no longer need this logic because the frontend now prevents segment deletion if the segment is being used in any surveys. To handle segment deletion through management APIs, we can simply check if the segment is associated with any surveys and, if it is, return an error message like "Deletion failed: Segment in use.". What do you think ? 🤔

Reply:
I think this check still makes sense here as the server action can simply be called from a curl request for example without involving the frontend, which can lead to problems I think, so a check in the service makes sense.

Reply:
Yes that is correct, we need that check but my concern is the logic that is used to loop over surveyIds and create new segments, do we still need that logic ?